File: src/main/java/com/javarush/island/shevchenko/entities/animals/herbivores/Duck.java
package com.javarush.island.shevchenko.entities.animals.herbivores;

import com.javarush.island.shevchenko.entities.organism.Animal;
import com.javarush.island.shevchenko.entities.organism.interfaces.*;
import com.javarush.island.shevchenko.settings.IslandSettings;
import com.javarush.island.shevchenko.util.Randomizer;

/**
 * Класс Duck представляет существо утка на острове животных.
 * Утки являются всеядными и используют поведение из интерфейса Omnivore для питания и охоты.
 */
public class Duck extends Animal implements Omnivore {

    public Duck() {
        this.aClass = Duck.class;
        this.weight = Randomizer.getRandom(IslandSettings.ANIMAL_PARAMETERS.get(Duck.class)[0] / 2, IslandSettings.ANIMAL_PARAMETERS.get(Duck.class)[0]);
    }
}

File: src/main/kotlin/ru/cooper/island/entities/animals/herbivores/Duck.kt
package ru.cooper.island.entities.animals.herbivores

import ru.cooper.island.entities.animals.herbivores.function.initAnimal
import ru.cooper.island.entities.organism.Animal
import ru.cooper.island.entities.organism.interfaces.Omnivore

class Duck : Animal(), Omnivore {
    init { initAnimal<Duck>() }
}
